mysql-test/suite/innodb/t/innodb-index.test

```diff
@@ -1193,6 +1193,14 @@ SHOW CREATE TABLE t;
 --disable_prepare_warnings
 DROP TABLE t;
 
+# Rename the index from lowercase to uppercase
+CREATE TABLE t1 (a INT,b INT,KEY(b)) ENGINE=INNODB;
+ALTER TABLE t1 RENAME KEY b TO B;
+SHOW CREATE TABLE t1;
+SELECT name FROM INFORMATION_SCHEMA.INNODB_SYS_INDEXES WHERE table_id IN (SELECT table_id FROM INFORMATION_SCHEMA.INNODB_SYS_TABLES WHERE NAME="test/t1");
```

Comment:
I think that we need to test also `ADD INDEX` and `DROP INDEX` where only the case of the name differs. I fear that we may cause a regression there, and the expectation should be that `DROP INDEX B` succeeds or `CREATE INDEX B` fails, no matter whether the key is named `b` or `B` in the data dictionary.
Case insensitive or sensitive comparisons for identifier names are something that should have been clarified in MySQL WL#922, which apparently never was implemented. That ticket is private, but a public comment in [MySQL Bug#35520](https://bugs.mysql.com/bug.php?id=35520) mentions it.